File: frontend/src/common/classes/Deck.ts
export default class Deck {
  id: number;

  name: string;

  description: string;

  isShared: boolean;

  constructor(
    id: number,
    name: string,
    description: string,
    isShared: boolean,
  ) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.isShared = isShared;
  }
}

File: backend/src/main/kotlin/ru/ivakin/mtgcollection/model/Deck.kt
package ru.ivakin.mtgcollection.model;

import org.springframework.data.relational.core.mapping.Table
import org.springframework.data.annotation.Id

@Table("deck")
data class Deck(
    @Id val id: Long?,
    val name: String,
    val description: String?,
    val is_chared: Int?
)
